timemachine/cpp/src/kernels/kernel_utils.cuh

```diff
@@ -32,6 +31,8 @@
 
 #define ONE_4PI_EPS0 138.935456
 
+static const int warp_size = 32;
```

Comment:
I learned that CUDA also defines a run-time constant `warpSize`, which is automatically correct for the device architecture (but this is always 32 on existing architectures). There's some conflicting advice [here](https://stackoverflow.com/questions/36047035/when-should-i-use-cudas-built-in-warpsize-as-opposed-to-my-own-proper-constant), but it seems like our approach of defining this as a compile-time constant might be better for our use case, since it allows for compiler optimization.